src/main/webapp/app/localci/build-queue/build-queue.component.ts

```diff
@@ -41,20 +123,49 @@ export class BuildQueueComponent implements OnInit, OnDestroy {
     ascending = false;
     interval: ReturnType<typeof setInterval>;
 
+    // Filter
+    @ViewChild('addressTypeahead', { static: true }) addressTypeahead: NgbTypeahead;
+    finishedBuildJobFilter = new FinishedBuildJobFilter();
+    faFilter = faFilter;
+    focus$ = new Subject<string>();
+    click$ = new Subject<string>();
+    isLoading = false;
+    search = new Subject<void>();
+    searchTerm?: string = undefined;
+
     constructor(
         private route: ActivatedRoute,
         private websocketService: JhiWebsocketService,
         private buildQueueService: BuildQueueService,
         private alertService: AlertService,
+        private modalService: NgbModal,
+        private localStorage: LocalStorageService,
     ) {}
 
     ngOnInit() {
         this.loadQueue();
         this.interval = setInterval(() => {
             this.updateBuildJobDuration();
         }, 1000);
-        this.loadFinishedBuildJobs();
         this.initWebsocketSubscription();
+        this.loadFilterFromLocalStorage();
+        this.loadFinishedBuildJobs();
+        this.search
+            .pipe(
+                debounceTime(UI_RELOAD_TIME),
+                tap(() => (this.isLoading = true)),
+                switchMap(() => this.FetchFinishedBuildJobs()),
+            )
+            .subscribe({
```

Comment:
Could you store this subscription and unsubscribe in `ngOnDestroy`? Many other components do this to prevent potential memory leaks.